From 648a2278a27610f85f05f15891b62f1b5cbd4550 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:36:50 -0400
Subject: [virt] kvm/iommu: Add leading zeros to device's BDF notation in debug messages

Message-id: <1406040016-3289-16-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86094
O-Subject: [RHEL7 PATCH v2 015/221] virt/kvm/iommu.c: Add leading zeros to device's BDF notation in debug messages
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Andre Richter <andre.o.richter@gmail.com>

When KVM (de)assigns PCI(e) devices to VMs, a debug message is printed
including the BDF notation of the respective device. Currently, the BDF
notation does not have the commonly used leading zeros. This produces
messages like "assign device 0:1:8.0", which look strange at first sight.

The patch fixes this by exchanging the printk(KERN_DEBUG ...) with dev_info()
and also inserts "kvm" into the debug message, so that it is obvious where
the message comes from. Also reduces LoC.

Acked-by: Alex Williamson <alex.williamson@redhat.com>
Signed-off-by: Andre Richter <andre.o.richter@gmail.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
(cherry picked from commit 29242cb5c63b1f8e12e8055ba1a6c3e0004fa86d)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/virt/kvm/iommu.c b/virt/kvm/iommu.c
index 0a54456..c7d9ce1 100644
--- a/virt/kvm/iommu.c
+++ b/virt/kvm/iommu.c
@@ -192,11 +192,7 @@ int kvm_assign_device(struct kvm *kvm,
 
  pdev->dev_flags |= PCI_DEV_FLAGS_ASSIGNED;
 
- printk(KERN_DEBUG "assign device %x:%x:%x.%x\n",
-  assigned_dev->host_segnr,
-  assigned_dev->host_busnr,
-  PCI_SLOT(assigned_dev->host_devfn),
-  PCI_FUNC(assigned_dev->host_devfn));
+ dev_info(&pdev->dev, "kvm assign device\n");
 
  return 0;
 out_unmap:
@@ -222,11 +218,7 @@ int kvm_deassign_device(struct kvm *kvm,
 
  pdev->dev_flags &= ~PCI_DEV_FLAGS_ASSIGNED;
 
- printk(KERN_DEBUG "deassign device %x:%x:%x.%x\n",
-  assigned_dev->host_segnr,
-  assigned_dev->host_busnr,
-  PCI_SLOT(assigned_dev->host_devfn),
-  PCI_FUNC(assigned_dev->host_devfn));
+ dev_info(&pdev->dev, "kvm deassign device\n");
 
  return 0;
 }
-- 
1.7.1